sqlmesh/cli/main.py

```diff
@@ -264,6 +264,12 @@ def diff(ctx: click.Context, environment: t.Optional[str] = None) -> None:
     help="Disable automatic change categorization.",
     default=None,
 )
+@click.option(
+    "--promote-all",
```

Comment:
what is a concrete use case of wanting to promote all ?

Reply:
Wanting to point dashboard(s) to an environment using a global find/replace instead of adjusting each reference that is included in the environment.

Reply:
Basically any use case where you want a complete representation of prod to validate against instead of just seeing a subset.

Reply:
i see, i was confused on the purpose of this. this is more important in the blank environment case

Reply:
> this is more important in the blank environment case

this is exactly it.

> once you have an environment and make a change, do we only ever promote things that have changed? eg there's no option for this one?

We only promote things that have changed + the things that were promoted in that environment previously. This is relevant to our other thread.


Additionally users can change their minds and apply a plan with `--promote-all` to an existing dev environment, in which case everything will be promoted.